vtex-apps/store-components
# Update: SKU Selector logic for default

**What problem is this solving?**
The SKU selector, when there were multiple SKUs with the same specifications, was always selecting the first SKU without considering the product’s sale feasibility. This issue could lead to displaying an SKU that is not actually available for purchase, affecting customer experience and conversions.

**How to test it?**
Ensure that you have multiple SKUs with identical specifications but differing in sale feasibility. Verify that the SKU selector now prioritizes SKUs that are available for sale when displaying the SKU options in the context.

[Workspace](https://selector--dunnesstores.myvtex.com/regular-fit-crew-neck-t-shirt-4541124dkna/p?skuId=20918)

**Screenshots or example usage:**

Before adjustment:
![image](https://github.com/user-attachments/assets/1a0245a1-5c68-4cfd-984f-77e76b1a589e)

After adjustment:
![image](https://github.com/user-attachments/assets/544f5f93-32b0-4743-a194-165b16b56469)

**Describe alternatives you've considered, if any.**
No significant alternatives were considered, as the focus was on improving SKU selection based on sale feasibility.

**How does this PR make you feel? [:link:](http://giphy.com/)**
![](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExd3Y4aXNheTg3MzlkanJyY3Azd3Z2eWt3dGNiaHY3enFjaG1seW11NiZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/3o9bJX4O9ShW1L32eY/giphy.gif)
